ObjectiveGit/GTBranch.m

```diff
@@ -193,7 +193,12 @@ - (GTBranch *)trackingBranchWithError:(NSError **)error success:(BOOL *)success
 }
 
 - (BOOL)updateTrackingBranch:(GTBranch *)trackingBranch error:(NSError **)error {
-	int result = git_branch_set_upstream(self.reference.git_reference, trackingBranch.shortName.UTF8String);
+  int result = GIT_ENOTFOUND;
+  if (trackingBranch.branchType == GTBranchTypeRemote) {
+    result = git_branch_set_upstream(self.reference.git_reference, [trackingBranch.name stringByReplacingOccurrencesOfString:[GTBranch remoteNamePrefix] withString:@""].UTF8String);
```

Comment:
Tracking branch could be either local branch or remote branch, you should have a look at [here](http://stackoverflow.com/questions/16408300/what-is-the-differences-between-local-branch-local-tracking-branch-remote-bran). Original method works perfectly when tracking a local branch, but not working correctly when tracking a remote branch because of the wrong parameter `shortName.UTF8String`